CriteoPublisherSdk/Sources/CR_BidManager.m

```diff
@@ -288,6 +289,30 @@ - (void)updateTimeToNextCallIfProvided:(CR_CdbResponse *)cdbResponse {
   }
 }
 
+- (void)enrichAdObject:(id)object withBid:(CRBid *)bid {
+  @try {
+    [self enrichUnsafelyAdObject:object withBid:bid];
+  } @catch (NSException *exception) {
+    CLogException(exception);
+  }
+}
+
+- (void)enrichUnsafelyAdObject:(id)object withBid:(CRBid *)bid {
+  if (!config) {
+    CLog(@"Config hasn't been fetched. So no bids will be fetched.");
```

Comment:
Bid is already fetched. I guess that such conditions are not valid anymore with the new logic. Same thing for the kill switch.

Moreover, I thought that bidding should not have to wait for the config to be fetched.

Reply:
> Bid is already fetched. I guess that such conditions are not valid anymore with the new logic. Same thing for the kill switch.

For some reason I was thinking the killswitch wasn't meant only for bid requests. Will change this.

> Moreover, I thought that bidding should not have to wait for the config to be fetched.

Not sure how this relates to this particular code as there is no wait there?
Are you talking about the bid fetching general case?

Reply:
> > Bid is already fetched. I guess that such conditions are not valid anymore with the new logic. Same thing for the kill switch.
> 
> For some reason I was thinking the killswitch wasn't meant only for bid requests. Will change this.

You're right, the kill switch is for deactivating the whole SDK. But IMHO just cutting the network+bidding logic (so `bid=nil`) is enough to deactivate the SDK.
So ok for keeping or removing the killSwitch check.

> 
> > Moreover, I thought that bidding should not have to wait for the config to be fetched.
> 
> Not sure how this relates to this particular code as there is no wait there?
> Are you talking about the bid fetching general case?

I was talking about the `if (!config)`. This condition should not belong to this method as you already have a bid. But I'm even questioning if it belongs to the `loadBid` method: bidding+config are async, so bidding should not care if config was fetched or not.

Reply:
Thanks for the details, will remove then as you said it is covered anyway on bid requesting side which is more central and logical.

Reply:
Done